src/tracing/internal/track_event_internal.cc

```diff
@@ -139,6 +143,18 @@ bool NameMatchesPatternList(const std::vector<std::string>& patterns,
   return false;
 }
 
+std::vector<const TrackEventCategoryRegistry*>&
+GetTrackEventCategoryRegistries() {
+  static base::NoDestructor<std::vector<const TrackEventCategoryRegistry*>>
+      registries;
+  return registries.ref();
+}
+
+std::mutex& GetTrackEventCategoryMutex() {
+  static std::mutex mu;
+  return mu;
+}
+
```

Comment:
Nit, optional:

Instead of having two static functions, can we have a single object, like

```
class RegistryState {
 public:
  RegistryState& GetInstance() {
    static base::NoDestructor<RegistryState> state;
    return state.ref();
  }
  
  // ...
  
  // ...

 private:
  std::mutex mu_;
  std::vector<>....
}
```

So that the mutex and the vector it protects are closer together? (I'm taking inspiration from perfetto::shlib::TrackEvent::GlobalState and  TrackEventSessionObserverRegistry). Maybe it makes less sense here, because we often have to use the mutex and the vector outside separately and we don't encapsulate all the behavior yet.